src/checker.c

```diff
@@ -505,7 +505,7 @@ Operand check_expr_compound(Checker *self, Expr *expr, Ty *wanted) {
                     index = (u32) op.val.i;
                 }
                 if (type && type->tarray.length && index >= type->tarray.length) {
-                    error(self, field.key->range,
+					error(self, (field.key ?: field.val)->range,
```

Comment:
This was crashing and I'm not really sure what was wrong, I added this and now I no longer get to this branch... Smells like memory corruption?